tests/test_events_api.py

```diff
@@ -0,0 +1,166 @@
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+        self.event_api.set_resources(["Yy", "ZZ"])
+        self.event_api.set_source("orgin")
+        self.event_api.set_bus_name("magic-bus")
+
+        self.event_api.send()
+
+        mock_send.put_events.assert_called_once_with(
+            Entries=[
+                {
+                    "Detail": '{"x": 1}',
+                    "DetailType": "MY_TEST_EVENT",
+                    "EventBusName": "magic-bus",
+                    "Resources": ["Yy", "ZZ"],
+                    "Source": "orgin",
+                }
+            ]
+        )
+
+    def test_register(self) -> None:
+        assert self.event_api.get_all_pending_events() == []
+
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        assert self.event_api.get_all_pending_events() == [event]
+
+    def test_get_all_registered_events(self) -> None:
+        assert self.event_api.get_all_pending_events() == []
+
+        event_1 = MyTestEvent({"x": 1})
+        event_2 = MyTestEvent({"x": 2})
+        self.event_api.register(event_1)
+        self.event_api.register(event_2)
+        assert self.event_api.get_all_pending_events() == [event_1, event_2]
+
+    @patch.object(Boto3Client, "eventbridge")
+    def test_send(self, mock_send: MagicMock) -> None:
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        self.event_api.send()
+
+        mock_send.put_events.assert_called_once_with(
+            Entries=[
+                {
+                    "Detail": '{"x": 1}',
+                    "DetailType": "MY_TEST_EVENT",
+                    "EventBusName": "million-dollar-lambda-event-bus",
+                    "Resources": [],
+                    "Source": "million-dollar-lambda",
+                }
+            ]
+        )
+        assert self.event_api.get_all_sent_events() == [event]
+
+    @patch.object(Boto3Client, "eventbridge", MagicMock())
+    def test_get_all_sent_events(self) -> None:
+        assert self.event_api.get_all_sent_events() == []
+
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        self.event_api.send()
+
+        assert self.event_api.get_all_sent_events() == [event]
+
+    @patch.object(Boto3Client, "eventbridge")
+    def test_get_all_failed_events(self, mock_send: MagicMock) -> None:
+        assert self.event_api.get_all_failed_events() == []
+
+        mock_send.put_events.side_effect = NotADirectoryError
+        event = MyTestEvent({"x": 1})
+        self.event_api.register(event)
+
+        with pytest.raises(NotADirectoryError):
+            self.event_api.send()
+
+        assert self.event_api.get_all_failed_events() == [event]
+
+    @patch.object(Boto3Client, "eventbridge")
+    def test_send_no_events(self, mock_send: MagicMock) -> None:
+        self.event_api.send()
+
+        mock_send.put_events.assert_not_called()
```

Comment:
You can also check that all the lists are empty, just to present the state more directly.